backend/asmgen.mli

```diff
@@ -94,7 +94,8 @@ val compile_unit
   Might return an instance of [Dwarf_ocaml.Dwarf.t] that can be used to generate
   dwarf information for the target system. *)
 val emit_begin_assembly_with_dwarf
```

Comment:
I find this signature a bit confusing.
`emit_begin_assembly_with_dwarf` is named as if calling it would turn on dwarf, but it also takes a `disable_dwarf` as input. 
Maybe it would be best to for `emit_begin_assembly_with_dwarf` to always setup dwarf and to have the caller not call it when we want dwarf off.

One other option could to be to change the signature of Emit.begin_assembly (or do a wrapper) with a flag `~turn_dwarf_on`

Reply:
Yeah, this isn't ideal, but refactoring this should wait until a subsequent PR.  I'm not sure offhand what the best option would be here.